### ISSUE TITLE
Request timeout should be disabled once connection comes back to the pool

### ISSUE THREAD

**Issue by carryel** [AsyncHttpClient/async-http-client#1109]
I am using AHC v1.9.32 with grizzly provider(grizzly framework v2.3.24).

And major configuration is the following:
- pooledConnectionIdleTimeout: 300000 (5min)
- requestTimeout: 1000 (1sec)
- allowPoolingConnections: true
- allowSslConnectionPools: true
- connectionTTL: -1

First, I expected http's idle-connection would be kept during 5min in pool but the connection would be always closed in about 1 sec.

When I debugged, grizzly provider used IdleTimeoutFilter with requestTimeout and the timeout was applied at the connection regardless of connection pooling. So the idle connection would be closed in requestTimeout.

Grizzly initializes the transport with IdleTimeoutFilter which has the request timeout.
When a connection has been established, the filter has the timer with FOREVER-timeout and the timer has the request timeout at next steps. When the connection has been closed, the filter resets the timer. 

But if we use connection pool, the connection is not closed until pool's idle timeout. So when the connection will be returned to the pool, the request timeout should be disabled.

**Comment by oleksiys** [comment]
Could you pls. create a simple testcase?

**Comment by carryel** [comment]
@oleksiys 
Sure. But.. AHC testcases can't get the client connection so you should test some manually. :-) 

1. In PerRequestTimeoutTest.java, add the following testcase.
```
    @Test(groups = { "standalone", "default_provider" })
    public void testGlobalIdleTimeout2() throws IOException {
        final int requestTimeout = 10 * 1000; // Reproduced, after 10sec, connection was closed
        //final int requestTimeout = 30 * 1000; // connection was still ESTABLISHED
        try (AsyncHttpClient client = getAsyncHttpClient(
                new AsyncHttpClientConfig.Builder().setRequestTimeout(requestTimeout).build())) {
            for (int i = 0; i < 2; i++) {
                Future<Response> responseFuture = client.prepareGet(getTargetUrl()).execute();
                Response response = responseFuture.get();
                assertNotNull(response);
                assertEquals(response.getResponseBody(), MSG + MSG);

                System.out.println("#####Please watch your established port(" + port1 + ")");
                TimeUnit.SECONDS.sleep(15);
            }
        } catch (InterruptedException e) {
            fail("Interrupted.", e);
        } catch (ExecutionException e) {
            fail("Timed out.", e);
        }
    }
```

2. Run this testcase with GrizzlyPerRequestTimeoutTest.

3. Check ESTABLISHED port for a while. This test runs two requests with the same client. The same connection should be used without requestTimeout.

PS) NettyPerRequestTimeoutTest worked fine.

**Comment by oleksiys** [comment]
@carryel could you pls. test it now?

**Comment by carryel** [comment]
@oleksiys Yes. It works fine. Thanks! [rev.088948a]

**Comment by carryel** [comment]
@slandelle Could I know the approximate release plan of AHC v1.9.x?

**Comment by slandelle** [comment]
closed by 088948a

**Comment by slandelle** [comment]
@carryel @oleksiys done
